src/node_errors.cc

```diff
@@ -1044,6 +1050,304 @@ void TriggerUncaughtException(Isolate* isolate, const v8::TryCatch& try_catch) {
+  }
+  return tmpl;
+}
+
+void DOMException::Initialize(Environment* env, v8::Local<v8::Object> target) {
+  env->SetConstructorFunction(
+      target,
+      "DOMException",
+      GetConstructorTemplate(env),
+      Environment::SetConstructorFunctionFlag::NONE);
+}
+
+void DOMException::RegisterExternalReferences(
+    ExternalReferenceRegistry* registry) {
+  registry->Register(New);
+  registry->Register(GetMessage);
+  registry->Register(GetName);
+  registry->Register(GetStack);
+  registry->Register(GetCode);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    const TransferData& transferData) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(env, obj, transferData);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    const std::string& message,
+    const std::string& name) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(env, obj, message, name);
+}
+
+BaseObjectPtr<DOMException> DOMException::Create(
+    Environment* env,
+    Local<Value> message,
+    const std::string& name) {
+  HandleScope scope(env->isolate());
+
+  Local<Function> ctor;
+  if (!GetConstructorTemplate(env)->GetFunction(env->context()).ToLocal(&ctor))
+    return BaseObjectPtr<DOMException>();
+
+  Local<Object> obj;
+  if (!ctor->NewInstance(env->context()).ToLocal(&obj))
+    return BaseObjectPtr<DOMException>();
+
+  return MakeBaseObject<DOMException>(
+      env,
+      obj,
+      message,
+      String::NewFromUtf8(env->isolate(), name.c_str()).ToLocalChecked());
+}
+
+void DOMException::New(const FunctionCallbackInfo<Value>& args) {
+  CHECK(args.IsConstructCall());
+
+  new DOMException(
+      Environment::GetCurrent(args),
+      args.This(),
+      args[0],
+      args[1]);
+}
+
+DOMException::DOMException(
+    Environment* env,
+    Local<Object> object,
+    Local<Value> message,
+    Local<Value> name)
+    : BaseObject(env, object),
+      message(env->isolate(), message),
+      name(env->isolate(), name),
+      stack(env->isolate(), MakeErrorStack(env, message, name)) {}
+
+DOMException::DOMException(
+    Environment* env,
+    Local<Object> object,
+    const std::string& message,
+    const std::string& name)
+    : BaseObject(env, object) {
+  Local<Value> message_value =
+      String::NewFromUtf8(env->isolate(), message.c_str()).ToLocalChecked();
```

Comment:
We can get rid of the calls to the `*Check*()` functions in the constructors and still signal to the caller if a failure occurs by using a factory function to construct these objects and return a nullish value when there is an exception. Such a thing is already implemented by the `MessagePort` class constructors: https://github.com/nodejs/node/blob/3f51f052ec805ecae77b7c9e7470c5c1f0b82f3b/src/node_messaging.h#L221